connectors/sources/confluence.py

```diff
@@ -442,6 +449,58 @@ async def fetch_attachments(
                     "url": attachment_url,
                 }, attachment["_links"]["download"]
 
+    async def search_by_query(self, query):
+        async for response in self.confluence_client.paginated_api_call(
+            url_name=SEARCH,
+            query=f"{query}&{SEARCH_QUERY}",
+        ):
+            results = response.get("results", [])
+            for entity in results:
+                # entity can be space or content
+                entity_details = entity.get(SPACE) or entity.get(CONTENT)
+
+                if (
+                    entity_details["type"] == "attachment"
+                    and entity_details["container"].get("title") is None
+                ):
+                    continue
+
+                document = {
+                    "_id": entity_details.get("id"),
+                    "title": entity.get("title"),
+                    "_timestamp": entity.get("lastModified"),
+                    "body": entity.get("excerpt"),
+                    "type": entity.get("entityType"),
+                    "url": os.path.join(
+                        self.confluence_client.host_url, entity.get("url")[1:]
+                    ),
+                }
+                download_url = None
+                if document["type"] == "content":
+                    document.update(
+                        {
+                            "type": entity_details.get("type"),
+                            "space": entity_details.get("space", {}).get("name"),
+                        }
+                    )
+
+                    if document["type"] == "attachment":
+                        container = entity_details.get("container", {}).get("type")
+                        document.update(
+                            {
+                                "size": entity_details.get("extensions", {}).get(
+                                    "fileSize"
+                                ),
+                                container: entity_details.get("container", {}).get(
+                                    "title"
+                                ),
+                            }
+                        )
```

Comment:
```suggestion
                    file_size =  entity_details.get("extensions", {}).get("fileSize")
                    title = entity_details.get("container", {}).get("title")
                    document.update(
                         {
                             "size": file_size,
                             "container": title
                         }
                     )
```

I think using variables would improve readability here. Also container should be in quotes, right?

Reply:
Agreed. I've updated with variable names to improve readability. Also for container variable I've renamed it to `container_type` as it denotes the type of container.